Game Coders/flappy_bird/barrier.pde

```diff
@@ -17,7 +17,22 @@ class Barrier {
     x -= barrier_speed;
   }
   
-  void show() {
+  //Detects Bird and Barrier Collisions
+  boolean hits()
```

Comment:
```suggestion
  boolean hits(Bird b)
```

Notice how we are using the bird in the statement below, which is required to detect the collision, hence the bird itself needs to be passed as a parameter to detect the collision.